Sources/App/Scenes/CarPlaySceneDelegate.swift

```diff
@@ -0,0 +1,261 @@
+import CarPlay
+import Communicator
+import HAKit
+import PromiseKit
+import Shared
+
+
+public protocol EntitiesStateSubscription {
+    func subscribe()
+    func unsubscribe()
+}
+
+@available(iOS 16.0, *)
+class CarPlayDelegate: UIResponder {
+    public static let SUPPORTED_DOMAINS_WITH_STRING = [
+        "button": L10n.Carplay.Labels.buttons,
+        "cover": L10n.Carplay.Labels.covers,
+        "input_boolean": L10n.Carplay.Labels.inputBooleans,
+        "input_button": L10n.Carplay.Labels.inputButtons,
+        "light": L10n.Carplay.Labels.lights,
+        "lock": L10n.Carplay.Labels.locks,
+        "scene": L10n.Carplay.Labels.scenes,
+        "script": L10n.Carplay.Labels.scripts,
+        "switch": L10n.Carplay.Labels.switches
+    ]
+    
+    public let SUPPORTED_DOMAINS = SUPPORTED_DOMAINS_WITH_STRING.keys
+    
+    private var MAP_DOMAINS = [
+        "device_tracker",
+        "person",
+        "sensor",
+        "zone"
+    ]
+    
+    private var interfaceController: CPInterfaceController?
+    private var filteredEntities: [HAEntity] = []
+    private var entitiesGridTemplate: EntitiesGridTemplate?
+    private var domainsListTemplate: DomainsListTemplate?
+    private var entitiesStateSubscribeCancelable: HACancellable?
+    private var serverObserver: HACancellable?
+    private var serverId: Identifier<Server>? {
+        didSet {
+            loadEntities()
+        }
+    }
+    
+    let prefs = UserDefaults(suiteName: Constants.AppGroupID)!
```

Comment:
use the available global for this, you don't need to create a new one here

Reply:
this is done already in Sources/App/AppDelegate.swift